### DISCUSSION TITLE
Page expired (419) after upgrade to Laravel 9.4.1  and Livewire 2.10.4

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#4799] [7. Bugs (archived)]
I've just upgraded to Laravel `9.4.1` and Livewire `2.10.4`. I didn't change anything else, and I now get the following native browser message on every Livewire request, both on Safari and Chrome. I have ran `php artisan optimize:clear` and closed and reopened the tab as I thought it might be something with the session getting 'stuck' after the upgrade.

<img width="457" alt="Screenshot 2022-03-12 at 18 12 41" src="https://user-images.githubusercontent.com/5268113/158031725-f30031d4-c247-468c-a12a-625ac1f78f5c.png">

The message seems a little different to what others have previously been reporting where it says 'due to inactivity'. That's not the case here.

**Update**
I don't know why but now the message does say 'due to inactivity'

<img width="455" alt="Screenshot 2022-03-12 at 19 58 02" src="https://user-images.githubusercontent.com/5268113/158031604-77cab6fc-4da3-42f3-a141-44732cf84bd2.png">

In the network tab I can see the Livewire request gives a 419. As indicated I tried `php artisan optimize:clear` (which is `cache:clear`, `route:clear`, `view:clear` and `config:clear`) and tried changing the value `SESSION_COOKIE` in `config/session` so I'm sure that it's refreshed.

**Comment by joshhanley** [comment]
@joostvanhoof hmm that's not good, definitely shouldn't be popping up on every request. Do you publish your Livewire javascript assets by any chance? If not, then you can ignore the next bit. If so, you will need to make sure you publish them again (should do it on every update), as we did make some changes.

We did update the message though, hence why you are seeing a different message.

Hope this helps!

**Comment by joostvanhoof** [comment]
Just checked and previously my composer.json had this:

```php
"post-autoload-dump": [
  "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
  "@php artisan package:discover --ansi",
]
```

which I've just updated to

```php
"post-autoload-dump": [
  "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
  "@php artisan package:discover --ansi",
  "@php artisan vendor:publish --force --tag=livewire:assets --ansi"
]
```

and then ran `composer update`. I've also ran `php artisan livewire:publish --assets` to be sure.

The error is still there however (it's the message without the 'due to inactivity'). @joshhanley you have access to the polar-bear repo that I have shared with you earlier. There is an `upgrade-to-laravel-9` branch where you can see this behaviour (go the the page 'Shoppen' at the top and click on any of the filters).

**Reply by joostvanhoof** [reply]
@joshhanley just checking if you had a time to take a look and any questions you have or info you need from me. As indicated you have access to the repo, feel free to try it there and curious to hear if you run into the same behaviour.

**Reply by joshhanley** [reply]
@joostvanhoof hey sorry, I have been off. I will try and have a look in the next couple of days and see if I can replicate it 🙂

**Reply by joostvanhoof** [reply]
@joshhanley no worries, hope you had a great time off 🌴 . Let me know if you have any questions!

**Reply by joostvanhoof** [reply]
@joshhanley curious if you had time to take a look! Any more info you need just let me know.

**Comment by ariaieboy** [comment]
hi i ran to this problem and i managed to fix it.
in my case browser blocks the session cookie because my `SESSION_DOMAIN` was incurrect please check your browser after first request and see that laravel session cookie is set correctly or not

**Reply by joostvanhoof** [reply]
I never had a `SESSION_DOMAIN` set in my `.ENV` file. After doing so the error is still there.

**Reply by joostvanhoof** [reply]
@ariaieboy I checked and if I remove the value it still sets the correct domain, and I still get the error... Any ideas?

**Comment by SimonErich** [comment]
I had the same issue and the only thing that worked for me was to add the `web` middleware to the routes.
Since I developed them in a package, I had this issue:
https://stackoverflow.com/a/64953004/2734492

**Comment by ValiNiculae** [comment]
I had the same issue but in my case i had a kubernetes setup with 2 pods and my SESSION_DRIVER was "file" which of course its a problem because the session is generated and stored on that container and when you switch the page the load balancer can send you to a different pod that has a different session :). This also applies for Docker setup with multiple php containers I guess. 

**Solution**: Use the **Redis Session driver** or as a last resort you can update the  VerifyCsrfToken class like so:

```
class VerifyCsrfToken extends BaseVerifier
{
  protected $except = [
    'livewire/*'
  ];
}
```
I don't recommend this btw :)

Hope this will help somebody :)

**Reply by info-aamroni** [reply]
Special thanks to @ValiNiculae 

```php
class VerifyCsrfToken extends Middleware
{
    /**
     * The URIs that should be excluded from CSRF verification
     * @var array<int, string>
     */
    protected $except = [
        'livewire/*'
    ];
}
```

**Reply by Eyadoo98** [reply]
it work fine thanks a lot

**Reply by Eyadoo98** [reply]
but it not make user as login auth

**Comment by emersonlsg10** [comment]
Works to me just doing this:

File: app.blade.php or sidebar.blade.php

`<meta name="csrf-token" content="{{ csrf_token() }}">`

**Comment by Timjini** [comment]
Hi, 
After trying multiple solutions, I found out that what worked for me is to store sessions like SESSION_DRIVER=file instead of database/redis. (check this in .env)

Depending on your needs, these solutions would work. Also, don't forget something obvious: the APP_URL.
